### ISSUE TITLE
feature request: url configuration for posts, pages, etc

### ISSUE THREAD

**Issue by chadhs** [cryogen-project/cryogen#83]
the idea i have is to support both specifying the root folder for page types as well as url path like this:

```
{
 :site-url           "http://site.ext/"
 :post-root          "posts"
 :post-url          "/articles"
 :tag-root           "tags"
 :page-root          "pages"
 :page-root          ""
 :blog-prefix        ""
}
```

The result would be a blog where the home page and all pages are under / in the public folder and posts are under /articles.

How I would personally configure it for my blog is for everything to be under /; as I like having http://site.ext/postname for better SEO.  This is currently how I have my Pelican blog configured.

Disclosure:
I'm a Clojure newbie, but I'd be willing to help with this feature request if you agreed that it would be a good configuration option.

**Comment by lacarmen** [comment]
I think it would be a good idea to let users specify an output folder for their posts/pages/tags. I've been pretty busy lately but if you'd like to work on it I can help you out. 

The key [here](https://github.com/cryogen-project/cryogen-core/blob/master/src/cryogen_core/compiler.clj#L114) would probably have to be changed to `:source-uri` and a `:dest-uri` key would have to be added to specify the output location. A bunch of other keys would have to be changed as well throughout the compiler.

**Comment by chadhs** [comment]
:+1: i'll certainly take a look as i have time.  i'd love to contribute.

**Comment by lacarmen** [comment]
I just remembered a similar issue is open here as well: cryogen-project/cryogen-core#32

**Comment by lacarmen** [comment]
Implemented in #96 and cryogen-project/cryogen-core#56
